src/custom/identity/identity.cc

```diff
@@ -154,138 +145,88 @@ Context::Init()
   return kSuccess;
 }
 
-int
-Context::Execute(
-    const uint32_t payload_cnt, CustomPayload* payloads,
-    CustomGetNextInputFn_t input_fn, CustomGetOutputFn_t output_fn)
+void
+Context::ExecutePayload(
+    CustomPayload& payload, CustomGetNextInputFn_t input_fn,
+    CustomGetOutputFn_t output_fn)
 {
-  for (uint32_t pidx = 0; pidx < payload_cnt; ++pidx) {
-    CustomPayload& payload = payloads[pidx];
+  for (uint32_t output_idx = 0; output_idx < payload.output_cnt; ++output_idx) {
+    if (payload.error_code != 0) {
+      break;
+    }
 
-    for (uint32_t output_idx = 0; output_idx < payload.output_cnt;
-         ++output_idx) {
-      if (payload.error_code != 0) {
-        break;
-      }
+    const char* output_cname = payload.required_output_names[output_idx];
+    const auto itr = copy_map_.find(output_cname);
+    if (itr == copy_map_.end()) {
+      payload.error_code = kRequestOutput;
+      break;
+    }
 
-      const char* output_cname = payload.required_output_names[output_idx];
-      const auto itr = copy_map_.find(output_cname);
-      if (itr == copy_map_.end()) {
-        payload.error_code = kRequestOutput;
+    const std::string& input_name = itr->second.input_name_;
+    const DataType datatype = itr->second.datatype_;
+
+    std::vector<int64_t> shape;
+    if (model_config_.max_batch_size() != 0) {
+      shape.push_back(payload.batch_size);
+    }
+    for (uint32_t input_idx = 0; input_idx < payload.input_cnt; ++input_idx) {
```

Comment:
The step of "given an input name, extract its shape (and content)" can probably be abstracted to a helper function in the SDK. This should be two functions, one function only operates on one payload while the other operates on multiple payloads for dynamic batching.

Reply:
I will look again. Originally I went down that route but found some of this code was not shared enough between the custom backends. Perhaps I can abstract some virtual methods for any customizations.

Reply:
Ok, I will move it up to execute and let the child class define if is per payload or batch. I did notice in other backend, this top level for loop doesn't exist.